### ISSUE TITLE
Suppress nullable warning expression doesn't suppress warnings from nullable `GetEnumeartor` reference in foreach

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#38233]
**Version Used**: 

```
Branch master (13 Aug 2019)
Latest commit a08336 by David:
Merge pull request #37916 from dibarbet/partial_load_str

Update partial load dialog string to better match task center.
```

**Steps to Reproduce**:

```cs
#nullable enable
using System.Collections;

class E : IEnumerable
{
  public int[]? arr;
  public IEnumerator? GetEnumerator() => arr?.GetEnumerator();
  void M()
  {
    var e = new E() { arr = new int[10] };
    foreach(var x in e!) { }
  }
}
```

**Expected Behavior**:
No warnings. Warnings on `e` are explicitly suppressed

**Actual Behavior**:
`CS8602: Dereference of a possibly null reference.` is reported for the usage of `e`

**Comment by RikkiGibson** [comment]
Note that the example above should also warn on the GetEnumerator implementation. It probably doesn't because the interface type being used is oblivious.

I feel it raises a question a bit of whether pattern-based GetEnumerator should warn if the return type is annotated e.g. `IEnumerable?` or `IEnumerable<string>?` but not `IEnumerable<string?>`. Since it seems like you can never safely 'foreach' on something with a nullable-returning GetEnumerator method.

I don't think this is very urgent, though.
